packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -0,0 +1,205 @@
+  Element _target;
+  final FileSelectorPluginTestOverrides _overrides;
+
+  bool get _hasTestOverrides => _overrides != null;
+
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<XFile> loadFile({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) async {
+    return (await _loadFileHelper(false, acceptedTypeGroups)).first;
+  }
+
+  /// Open file dialog for loading files and return a XFile
+  @override
+  Future<List<XFile>> loadFiles({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String confirmButtonText,
+  }) async {
+    return _loadFileHelper(true, acceptedTypeGroups);
+  }
+
+  @override
+  Future<String> getSavePath({
+    List<XTypeGroup> acceptedTypeGroups,
+    String initialDirectory,
+    String suggestedName,
+    String confirmButtonText,
+  }) async =>
+      null;
+
+  /// Load Helper
+  Future<List<XFile>> _loadFileHelper(
+      bool multiple, List<XTypeGroup> acceptedTypes) {
+    final acceptedTypeString = _getStringFromFilterGroup(acceptedTypes);
+
+    final FileUploadInputElement element =
+        createFileInputElement(acceptedTypeString, multiple);
+
+    _target.children.clear();
+    addElementToContainerAndClick(_target, element);
+
+    return getFilesWhenReady(element);
+  }
+
+  /// Default constructor, initializes _target to a DOM element that we can use
+  /// to host HTML elements.
+  /// overrides parameter allows for testing to override functions
+  FileSelectorPlugin({
+    @visibleForTesting FileSelectorPluginTestOverrides overrides,
+  }) : _overrides = overrides {
+    _target = ensureInitialized(_kFileSelectorInputsDomId);
+  }
+
+  /// Registers this class as the default instance of [FileSelectorPlatform].
+  static void registerWith(Registrar registrar) {
+    FileSelectorPlatform.instance = FileSelectorPlugin();
+  }
+
+  void _verifyXTypeGroup(XTypeGroup group) {
+    if (group.extensions == null &&
+        group.mimeTypes == null &&
+        group.webWildCards == null) {
+      StateError(
+          "This XTypeGroup does not have types supported by the web implementation of loadFile.");
+    }
+  }
+
+  /// Convert list of XTypeGroups to a comma-separated string
+  String _getStringFromFilterGroup(List<XTypeGroup> acceptedTypes) {
+    List<String> allTypes = List();
+    for (XTypeGroup group in acceptedTypes ?? []) {
+      _verifyXTypeGroup(group);
+
+      for (String mimeType in group.mimeTypes ?? []) {
+        allTypes.add(mimeType);
+      }
+      for (String extension in group.extensions ?? []) {
+        String ext = extension;
+        if (ext.isNotEmpty && ext[0] != '.') {
+          ext = '.' + ext;
+        }
+
+        allTypes.add(ext);
+      }
+      for (String webWildCard in group.webWildCards ?? []) {
+        allTypes.add(webWildCard);
+      }
+    }
+    return allTypes?.where((e) => e.isNotEmpty)?.join(',') ?? '';
```

Comment:
Instead of having a `_verifyXTypeGroup` at the top, I'd do:

```dart
assert(allTypes.isNotEmpty, 'At least one of extensions / mimeTypes / webWildCards is required!');
```